pip_faster.py

```diff
@@ -41,209 +47,149 @@
         import pkg_resources
 
 
-def ignorecase_glob(glob):
-    return ''.join([
-        '[%s%s]' % (char.lower(), char.upper())
-        if char.isalpha() else
-        char
-        for char in glob
-    ])
+# Thanks six!
+PY2 = str is bytes
+if PY2:  # :pragma:nocover:
+    _reraise_src = 'def reraise(tp, value, tb=None): raise tp, value, tb'
+    exec(_reraise_src)  # pylint:disable=exec-used
+else:  # :pragma:nocover:
+    def reraise(tp, value, tb=None):  # pylint:disable=invalid-name
+        if value is None:
+            value = tp()
+        if value.__traceback__ is not tb:
+            raise value.with_traceback(tb)
+        raise value
 
 
-def is_local_directory(url):
-    """Test if a URL is a directory on this machine.
+class CacheOpts(object):
 
-    >>> is_local_directory('file:///tmp/')
-    True
-    >>> is_local_directory('file:///etc/passwd')
-    False
-    >>> is_local_directory('http://yelp.com/')
-    False
-    >>> is_local_directory('file:.')
-    True
-    """
-    if url is None:
-        return False
-    elif not url.startswith('file:'):
-        return False
-    else:
-        from os.path import isdir
-        return isdir(url[5:])
+    def __init__(self):
+        self.pipdir = user_cache_dir() + '/pip-faster'
+        self.wheelhouse = self.pipdir + '/wheelhouse'
 
 
-def is_vcs_url(url):
-    """Test if a URL is a VCS repo.
+def optimistic_wheel_search(req, index_urls):
+    name = req.name.replace('_', '-').lower()
 
-    >>> is_vcs_url('git+git://git.myproject.org/MyProject#egg=MyProject')
-    True
-    >>> is_vcs_url('svn+http://svn.myproject.org/svn/MyProject/trunk@2019#egg=MyProject')
-    True
-    >>> is_vcs_url('file:///tmp/')
-    False
-    """
-    if url is None:
-        return False
-    else:
-        return url.startswith(tuple(
-            scheme + '+'
-            for scheme in pipmodule.vcs.vcs
-        ))
-
-
-def optimistic_wheel_search(req, find_links):
-    assert req_is_pinned(req), req
-
-    # this matches the name-munging done in pip.wheel:
-    reqname = req.project_name.replace('-', '_')
-    reqname = ignorecase_glob(reqname)
-    reqname = reqname + '-*.whl'
-
-    for findlink in find_links:
-        if findlink.startswith('file:'):
-            findlink = findlink[5:]
-        from os.path import join
-        findlink = join(findlink, reqname)
-        logger.debug('wheel glob: %s', findlink)
-        from glob import glob
-        for link in glob(findlink):
-            from pip.index import Link
-            link = Link('file:' + link)
-            from pip.wheel import Wheel
-            wheel = Wheel(link.filename)
-            logger.debug('Candidate wheel: %s', link.filename)
-            if wheel.version in req and wheel.supported():
-                return link
+    if len(index_urls) != 1:
+        return
+
+    index_url, = index_urls
+    expected_location = os.path.join(
+        CacheOpts().wheelhouse,
+        index_url,
+        name,
```

Comment:
let's get rid of this path segment and restore the `--find-links` in venv-update (such that venv-update could be no-network)